### ISSUE TITLE
Strip auth not working with version 1.0.2

### ISSUE THREAD

**Issue by Maxyme** [conda-incubator/conda-lock#174]
Running this 
`conda-lock --lockfile test.lock --strip-auth -f environment.yml -p linux-64`
copies the authentication username and password from ~/.condarc into test.lock instead of stripping it out like it does with the explicit (old) kind.

**Comment by mariusvniekerk** [comment]
Your lock files should not be getting anything from condarc ever. How are you specifying your channels?

**Comment by Maxyme** [comment]
I'm using a custom JFROG channel for conda, specified like this in the environment.yml file:

`name: env
channels:
  - custom-channel
  - conda-forge
  - defaults
dependencies:`

then, in the .condarc, I have it as a custom channel:

```
auto_activate_base: false
channel_priority: disabled

channels:
  - conda-forge
  - default

default_channels:
  - conda-forge
  - default
custom_channels:
  custom-channel: https://USERNAME:PASSWORD@REPO.jfrog.io/artifactory/api/conda/
```

**Comment by jamesp** [comment]
@Maxyme have you considered using a netrc file to store the authentication credentials independently of your condarc file?  We have moved to this as good practise as it also stops your api key from leaking into other files such as `conda-meta`.

I have raised it with conda here: https://github.com/conda/conda/issues/1784#issuecomment-839879431. This seems to be an undocumented but very useful feature of conda.
